workers/unity/Assets/Playground/Scripts/MonoBehaviours/PlayerColor.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Generated.Playground;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEngine;
+using Color = UnityEngine.Color;
+
+namespace Playground.MonoBehaviours
+{
+    public class PlayerColor : MonoBehaviour
+    {
+        private static MaterialPropertyBlock basicMaterial;
+        private static MaterialPropertyBlock runningMaterial;
+        private static MaterialPropertyBlock flashingMaterial;
+
+        [RuntimeInitializeOnLoadMethod]
+        public static void SetupColors()
+        {
+            basicMaterial = new MaterialPropertyBlock();
+            basicMaterial.SetColor("_Color", Color.white);
+
+            flashingMaterial = new MaterialPropertyBlock();
+            flashingMaterial.SetColor("_Color", Color.red);
+
+            runningMaterial = new MaterialPropertyBlock();
+            runningMaterial.SetColor("_Color", Color.yellow);
+        }
+
+        [SerializeField] private float flashDurationSeconds = 0.05f;
+
+        [Require] PlayerInput.Requirables.Reader reader;
+
+        private MeshRenderer renderer;
+        private ILogDispatcher logger;
+        private bool isRunning = false;
+        private bool isFlashing = false;
+        private float flashStartTime;
+
+        void OnEnable()
+        {
+            if (reader != null) // TODO remove in UTY-791
+            {
+                reader.ComponentUpdated += OnUpdateReceived;
+                reader.RunningUpdated += OnRunningChanged;
+            }
+        }
+
+        void OnDisable()
+        {
+            if (reader != null) // TODO remove in UTY-791
+            {
+                reader.ComponentUpdated -= OnUpdateReceived;
+                reader.RunningUpdated -= OnRunningChanged;
+            }
+        }
+
+        private void Awake()
+        {
+            logger = GetComponent<SpatialOSComponent>()?.LogDispatcher;
+            renderer = transform.Find("Shape")?.GetComponent<MeshRenderer>();
+            if (logger == null)
+            {
+                Debug.LogError("PlayerColor behaviour on GameObject that is not connected to Spatial systems properly");
+            }
+            else if (renderer == null)
```

Comment:
It's `else if` so that I don't have to handle the logger not existing - I could add a case where I log the renderer error to Debug.LogError but I'm not sure it's worth the extra case handling for that

Reply:
Maybe - the current implementation means you might need to run twice to find and fix all the errors. Its no biggie - but could be a minor annoyance

Reply:
True, but it seems sufficiently unlikely for both to happen that I feel the small mental overhead of reading through two kinds of logging would be a tradeoff for the worse. It's not a strong conviction though, so say if you disagree and I'll change it